src/Symfony/Component/Routing/Loader/Configurator/Traits/LocalizedRouteTrait.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader\Configurator\Traits;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+trait LocalizedRouteTrait
```

Comment:
shouldn't all these traits be marked as internal?

Reply:
There are used in the framework bundle and should in any extension. I guess not

Reply:
yes but the implementation as a trait is an implementation detail and the trait should not be reused by users

Reply:
I think they should if they need to add custom loaders/configurators as done here

Reply:
If users add custom configurators, they would use different method with other return types (like this PR shows) which would not use those traits but only the classes.

Reply:
Yes I agree that it would be better to extend configurators that use the traits directly. But if one want to create a `JsonLoader` or any other type it would be nice to use the traits as done here for YAML and XML but the I created this trait to also "reuse" its method in the new configurator. Maybe we could have an `AbstractRoutingLoader` using the traits and extended by the YAML and XML one, so users can extend it too instead or using the traits directly?

Reply:
@Tobion see #35731.